### ISSUE TITLE
Rolling out Twitch adblocking in Brave

### ISSUE THREAD

**Issue by ryanbr** [pixeltris/TwitchAdSolutions#282]
Just a heads up,

In the process of rolling out twitch adblocking https://github.com/brave/brave-core/pull/25546

Using https://github.com/brave/adblock-lists/blob/master/brave-lists/brave-twitch.txt  (Scriplet: https://github.com/brave/adblock-resources/blob/master/resources/vaft-ublock-origin.js) 

I haven't seen any reports of broken issues other than https://old.reddit.com/r/brave_browser/comments/1f7ny6s/twitchtv_doesnt_work_anymore/  But I don't believe its an issue with the scriplet here.

Want to test it to be sure it works well? Enable `Brave experimental Adblock Rules` in `brave://settings/shields/filters `

Does it work well?

**Comment by stevenya97** [comment]
Likely conflicting with AdGuard Extra's vaft script. Vaft will switch to a lower quality stream with no ads during midrolls, as the normal stream will have the full screen ad, and switch back once midroll is over. Potentially this addon setting is breaking this core function and causing it to buffer. Would have to look at the code to be sure, but it not happening with the setting disable certainly suggests so. Could be something to note in this repo's README down the line if more people come across the same issue.

**Comment by ryanbr** [comment]
@pixeltris is there a downside to https://github.com/pixeltris/TwitchAdSolutions/pull/199 ?

**Comment by ryanbr** [comment]
@kodiakhub  I don't think we can avoid issues with 3rd party twitch scripts/extensions. Similar to the issues people see with Youtube ads, list authors don't recommend using privacy/YT/other adblock extensions which may trigger false positives.

**Comment by ryanbr** [comment]
We'll take it on board, though users probably won't notice it unless its pointed out (via a support agent)

**Comment by ryanbr** [comment]
@pixeltris Getting some reports;

```
video player not loading. applies to every stream i visit
Twitch broadcasts are not opening, I tried to delete cookies again.
twitch live videos/vods doesn't work with brave Shields on for some reason
twitch won't play (took forever to load) the video without disabling shield
```

**Comment by ryanbr** [comment]
Trying to reach to one user, in case its an extension compatibility issue.

**Comment by ryanbr** [comment]
https://old.reddit.com/r/brave_browser/comments/1fjqmd8/brave_shield_no_longer_working_on_twitch/
https://old.reddit.com/r/brave_browser/comments/1fjrhxl/brave_twitch_adblock_rules_list_has_started_to/

Asking about extensions, my hunch

**Comment by pixeltris** [comment]
No playback / infinite loading wheel means conflicting solutions (Worker hooked multiple times). There's a list of chromium extensions here which all use an older copy of `vaft` https://github.com/pixeltris/TwitchAdSolutions/blob/master/full-list.md#extensions-to-avoid. Similarly @arthurbolsoni's [Purple Adblock](https://chromewebstore.google.com/detail/purple-ads-blocker/lkgcfobnmghhbhgekffaadadhmeoindg). Someone could also have an old `vaft` / `video-swap-new` permalink hooked up to Brave which would conflict. There probably are more.

> @pixeltris is there a downside to https://github.com/pixeltris/TwitchAdSolutions/pull/199 ?

This PR is specifically for improving freezing issues which is a separate thing from the no playback / infinite loading wheel. The Worker hook check could probably be improved such as preventing overwriting (i.e. the reverse of #169), though this may cause other problems.

> can’t use the redeem function or whispers or even subscribe to people. I moderate a community as well and I can’t do any of the mod actions

I'm not too sure about this one as I haven't tested anything with an account in a long time. There's a similar thing with clip creation #121

**Comment by ShivanKaul** [comment]
Is there any way we can detect in the JS that there's an older copy of `vaft` being used, and in that case, not inject or try to register the Worker?

**Comment by pixeltris** [comment]
Specifically `vaft` there are a bunch of variables names which get set up in the `window` context inside of: https://github.com/pixeltris/TwitchAdSolutions/blob/859802410009e3a9bf47a804a502f2dbbf38d150/vaft/vaft-ublock-origin.js#L4

But if you check for something like this you'd need to make sure the older script loads first.

**Comment by ShivanKaul** [comment]
> If another script comes along which also uses Object.defineProperty as assignment and doesn't check the integrity of the Worker you'll run into the same problems.

I think at some point we'll just have to tell users that they shouldn't be installing janky extensions. It's the first roll out that's the tricky thing.

**Comment by ShivanKaul** [comment]
If a user installs an extension that immediately starts causing issues, then they know that it's probably the new extension that's at fault. Currently however, given that we're deploying this new script out, it looks like [TwitchAdSolutions/vaft/vaft.user.js](https://github.com/pixeltris/TwitchAdSolutions/blob/859802410009e3a9bf47a804a502f2dbbf38d150/vaft/vaft.user.js#L869) is breaking things, not the older and worse scripts.

**Comment by ryanbr** [comment]
Just to outline we reverted the twitch scriptlet due to the breakages from other extensions, When we test this I had no issues it did a good job at block the adverts. 

But I wasn't using any 3rd party twitch extensions either. Having the script exit gracefully if detecting an extension or other methods would be ideal tbh. From what was discussed from the community these 2 extensions came up:

- Stream Cleaner
- Purple Ads Blocker

**Comment by pixeltris** [comment]
As mentioned above using `Object.defineProperty` / `writable:false` to assign the Worker is the only way I can see to prevent the double Worker hook with the current versions of the extensions. I can't say what side effects that may have, but it does seem to work from my limited testing.

Alternatively you'd need to do something on the Brave side of things to check for installed extensions as the script doesn't have access to this kind of information.

**Comment by pixeltris** [comment]
https://github.com/pixeltris/TwitchAdSolutions/commit/69f098d882eba30a6466ad99eadf487c0aa68139 should fix it. This checks if other scripts / extensions have modified the Worker after the script does. With the current versions of `Stream Cleaner` / `Purple Ads Blocker` it'll always use their extension rather than the script.

**Comment by ryanbr** [comment]
Asking if the user has any extensions installed; Not sure if you've heard of this issue.

https://x.com/CryptoWaDummy/status/1839879698509074786

`I was using the Content filter "Brave Twitch Adblock rules" while I was in a streamers chat it would say it was in emote only. In reality it was not. When I disabled the filter it worked as intended`

**Comment by ryanbr** [comment]
https://github.com/brave/brave-browser/issues/22370#issuecomment-2464279890

Some initial testing with various extensions (old and new)

**Comment by pixeltris** [comment]
I see with the recent change https://github.com/pixeltris/TwitchAdSolutions/commit/722f5b96241ef1d92998b87f941690e47e475379 there's a regression with the script + TTV LOL PRO which results in neither script being applied. @younesaassila

`Stream Cleaner` does work from my testing, both alone and with the experimental Brave filter.

`Better Twitch Adblock` / `Adblock for Twitch` are broken as you mention as they are outdated / lack this fix https://github.com/pixeltris/TwitchAdSolutions/commit/044d1fb3bb88a19ab718603069354d43eeba7015. But it's still triggering the Brave script's check for another active solution so ads get through.

Not ideal. I'll have a think about the conflicts with broken solutions.

**Comment by ryanbr** [comment]
Its better than breaking the site to be fair (the endless black screen looping issues previously), but would be nice if interactions between TwitchAdSolutions and some extensions could be tweaked/improved. 

Thanks for updates and contributions. Much appreciated

**Comment by pixeltris** [comment]
Try out this version https://gist.github.com/pixeltris/8c40313fcc8da2687381ba7b6c10adff (diff: https://gist.github.com/pixeltris/8c40313fcc8da2687381ba7b6c10adff/revisions)

It should always block ads and there shouldn't be conflicts with the extensions you tested.

It has the following properties:
- It hooks the lowest level non-twitch related `Worker` instance
- It assigns `window.Worker` with `Object.defineProperty` / `writable:false` to prevent hooking over our `Worker` hook
- If a solution manages to hook over our `Worker` hook we detect it and pull out the original url the `Worker` was initialized with. If it fails to find the original url it does an `eval` to prevent the black screen / infinite loading screen
- I added a version number to the script in `window` scope which can be bumped when conflicts arise with extensions which copy/paste the script and leave it in a broken / unmaintained state

Problems:
- The method of preventing other things from hooking `Worker` may cause problems for extensions which depend on hooking `Worker`
- The worker `onmessage` is being overridden by other extensions when they are applied after the script. **TODO:** Switch to using `addEventListener`, modify all currently used message names to avoid conflicts with old solutions, and add some validation that the script's hook is active
- If an extension copies the script or the follows the exact same logic then the solution that will be applied will be the one which is injected last which will be inconsistent

**Comment by ryanbr** [comment]
Nice. can you update the *-ublock-origin.js ? How safe is it to roll out for testing in Brave?

**Comment by ryanbr** [comment]
Any updates @pixeltris ?

**Comment by pixeltris** [comment]
Updated the script and addressed the above problems.

The script now removes already applied Twitch ad blocking scripts from the Worker hierarchy and uses a getter/setter to allow extensions like https://github.com/besuper/TwitchNoSub to still function while still preventing Twitch ad blocking solutions from breaking anything.

**Comment by pktpls** [comment]
Saw my first ad in months a few minutes ago, any chance this is related to the changes here, or did Twitch just deploy other changes?

**Comment by ryanbr** [comment]
Twitch couldn't change/add checks on `window.ourTwitchAdSolutionsVersion` ?

**Comment by pixeltris** [comment]
They could, but there are other things they could already check for such as the integrity of Worker and this piece of code which has been around for a while:

https://github.com/pixeltris/TwitchAdSolutions/blob/1c41aeab0d8922e96da62903896f2785c169e932/vaft/vaft-ublock-origin.js#L807

**Comment by Nemisor** [comment]
![{FD8C964E-5CF1-4B40-9990-739A25DCCBFA}](https://github.com/user-attachments/assets/bd303354-5d30-417a-b38d-e6253ad7ce2e)
Just to make sure, this is only the Experimental Rules, and we should or should not enable the Twitch Rules as well?

**Comment by ryanbr** [comment]
Currently only in Experimental, not updated yet to the latest ver. But still works.

**Comment by ryanbr** [comment]
Can TwitchAdSolutions conflict with itself, if running in uBO and Brave, or Brave + greasemonkey?

**Comment by pixeltris** [comment]
Nope. The following prevents double script loading:

https://github.com/pixeltris/TwitchAdSolutions/blob/ea9a01eb455e927f29ba711e881451784a8508ec/vaft/vaft.user.js#L16-L21

Which means the script which will be used will be the script which loads first.

If a project forks the code and removes that check then the script which will be used will be the script which loads last (as it will overwrite the Worker hook). But it currently wont cause a conflict in terms of playback problems (this may change if a fork which removes the above check is not properly maintained).

https://github.com/pixeltris/TwitchAdSolutions/blob/ea9a01eb455e927f29ba711e881451784a8508ec/vaft/vaft.user.js#L114-L115
https://github.com/pixeltris/TwitchAdSolutions/blob/ea9a01eb455e927f29ba711e881451784a8508ec/vaft/vaft.user.js#L287-L299

**Comment by ryanbr** [comment]
I haven't check but incompatibilities with a user using AdguardExtra?  https://old.reddit.com/r/brave/comments/1jiajz1/twitch_ads_has_anyone_found_a_way_to_block_them/

Link: https://github.com/AdguardTeam/AdGuardExtra

**Comment by ryanbr** [comment]
Wanting to roll out twitch adblocking as default soonish in Brave, does https://github.com/pixeltris/TwitchAdSolutions/issues/345 have any issues? any upcoming improvements? @pixeltris

**Comment by pixeltris** [comment]
> does https://github.com/pixeltris/TwitchAdSolutions/issues/345 have any issues?

From the videos it kind of looks like client side ads? Probably preventable using a filter? Not sure about the unrelated playback issue mentioned in there.

> any upcoming improvements?

I haven't personally been using Twitch much lately so I've been neglecting stuff here. #336 needs fixing but it might only impact `video-swap-new`? Other than that I think the only other change would be possibly getting in some version of #313 at some point to help with infinite buffering issues.

**Comment by mogzol** [comment]
@ryanbr So is this enabled by default in Brave now? It seems like it is, because vaft appears to be running and I don't have the "Brave Twitch Adblock Rules" or "Brave Experimental Adblock Rules" lists enabled. Is there any way to disable this behaviour? vaft breaks TTV LOL PRO which works better for me (I have a custom proxy set up for it). I know I can turn off Brave shields for twitch, but is there any way to just turn off vaft without disabling all of Brave's shields?

**Comment by ryanbr** [comment]
Shouldn't break anything @mogzol . @pixeltris rolled out updates to avoid incompatibilites. I'll re-check again, but last time I tested TTV LOL Pro, it didn't break

**Comment by mogzol** [comment]
@ryanbr When brave shields are turned on, TTV LOL PRO doesn't proxy any requests, and checking the console logs, I never get the `[TTV LOL PRO] Worker script running.` log message. When brave shields are off, it works correctly.

It does work properly if I change TTV LOL PRO's settings to "Proxy all requests" instead of "Proxy ad requests only", but I'd rather not have to do that.

EDIT: The issue seems to be that TTV LOL PRO attempts to set `window.Worker` here: https://github.com/younesaassila/ttv-lol-pro/blob/v2.4.0/src/page/page.ts#L46

But the vaft script prevents this from happening, here: https://github.com/pixeltris/TwitchAdSolutions/blob/v2/vaft/vaft-ublock-origin.js#L275

**Comment by ryanbr** [comment]
Okay, adding `twitch.tv#@#+js(vaft-ublock-origin)` in brave://adblock would get around this. If there is a possible work around here @pixeltris ?

**Comment by ryanbr** [comment]
From my testing earlier

https://github.com/brave/brave-browser/issues/22370#issuecomment-2464279890
https://github.com/brave/brave-browser/issues/22370#issuecomment-2612275395

Maybe Twitch ads have changed since then to affect TTV Lol Pro. Better than a broken video (the previous patch)

**Comment by mogzol** [comment]
For some reason adding `twitch.tv#@#+js(vaft-ublock-origin)` didn't disable vaft. I even tried disabling all scriptlets with `twitch.tv#@#+js()` as described [here](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#scriptlet-injection), but vaft was still running.

I ended up creating a custom scriplet to reset window.Worker, here it is if anyone else wants it:

1. Go to brave://settings/shields/filters
2. Enable developer mode
3. Add a new scriptlet in the "Custom scriptlets" section named `user-vaft-disable.js` with this content:
   ```js
   let nativeWorker = window.Worker;
   while (!nativeWorker.toString().includes("[native code]")) {
     nativeWorker = Object.getPrototypeOf(nativeWorker);
   }
   
   delete window.Worker;
   window.Worker = nativeWorker;
   ```
4. Enable the scriptlet for twitch.tv by adding this into the "Create custom filters" section:
   ```
   twitch.tv##+js(user-vaft-disable)
   ```
5. Click "Save changes" and then try opening twitch. Vaft's worker should be disabled and TTV LOL PRO should work.

I don't know if this has any side effects since it doesn't fully disable vaft, just clears out the changes to the worker, but it seems to work fine.

**Comment by ryanbr** [comment]
See if this helps, https://github.com/brave/adblock-lists/pull/2515
Moved to first-party list to allow exception to work

**Comment by mogzol** [comment]
After updating my filter lists, `twitch.tv#@#+js(vaft-ublock-origin)` works now to disable vaft. Thank you!

**Comment by zGato** [comment]
Considering how conflicting and how Twitch is about adblockers, making this not even a toggle off in the Shields options is just insane to me.

And yeah, this update breaks TTV LOL PRO since we can't inject properly, using the default proxy option that we also recommend, since proxy all uses an insane amount of bandwidth and resources.
![Image](https://github.com/user-attachments/assets/6769df9a-3cf8-4220-8bf8-db027c1f4b1b)

Not even disabling Shields works.
https://img.perfprod.com/files/iGx3ogxrTHG6.gif

Either way, I've already received multiple reports from other users using Brave and TTV LOL PRO, and personally adding a warning to not use Brave until this is fixed. 

This is also a great way to kill an adblock solution btw :)

**Comment by mogzol** [comment]
That's weird, disabling shields definitely fixed it for me (or adding that line to my filter lists to just disable vaft). But yeah I agree, making it so difficult to disable the script seems like a big oversight when a lot of people use other twitch ad blocker solutions which are now just broken by default on Brave.

**Comment by antonok-edm** [comment]
it's been tested with several Twitch-related extensions; TTV LOL PRO seems to be doing something differently than the rest that isn't working. We can look for a way to keep both active together but in the meantime the custom rule above should help.

**Comment by zGato** [comment]
> it's been tested with several Twitch-related extensions; TTV LOL PRO seems to be doing something differently than the rest that isn't working. We can look for a way to keep both active together but in the meantime the custom rule above should help.

"it's been tested with several Twitch-related extensions" you mean all vaft/video-swap-new wrappers, besides other Twitch extensions **that are not focused on ad-blocking**?. 
[Stream Cleaner](https://chromewebstore.google.com/detail/stream-cleaner/lehcglgkjkamolcflammloedahjocbbg)
[Adblock for Twitch](https://chromewebstore.google.com/detail/adblock-for-twitch/mdomkpjejpboocpojfikalapgholajdc)

are both video-swap-new/vaft wrappers.

This is not even about making them compatible. We've already tried, and it's a whole mess that is not even worth it. This is about how Brave is now enabling **by default to everyone** a feature that **cannot be disabled** and breaks third party extensions.

I'll try to dig into it a bit deeper, but apparently requests do still get proxied, but there's some conflict and the extension doesn't get to count them as proxied requests, nor we can identify the proxy used since the DNS part also seems to fail? (of course, only when Shields is disabled, otherwise it's just broken)

**Comment by ryanbr** [comment]
https://github.com/brave/brave-browser/issues/22370#issuecomment-2464279890

Was well tested, we tested the roll out a few patches previously but broke the video entirely. The current scriptlet seems good/much better.  Most people using 3rd-party twitch extensions will use it to counter ads, there isn't really a need for them anymore. Other features? sure!

**Comment by mogzol** [comment]
> There isn't really a need for them anymore.

I disagree with this. vaft works but it is (imo) far worse than a proxy-based solution like TTV LOL PRO. It has to switch to a lower quality video when blocking ads, which sometimes doesn't work and has known issues (#336 is one that really annoyed me). Proxy-based ad block gives you the full quality stream all the time without swapping and has, in my experience, always worked much better.

**Comment by ryanbr** [comment]
I was actually looking into this last night https://github.com/pixeltris/TwitchAdSolutions/issues/336

The pause only occurred for me when the stream was muted. The moment the stream was playing back audio (set the volume), I was able to switch tabs and the orginal tab kept playing in the background.

**Comment by mogzol** [comment]
For me it would pause whenever the tab was unfocused and it tried to block ads. But maybe that's a separate issue to #336, idk

**Comment by ryanbr** [comment]
@pixeltris Twitch has changed the blocking, got reports of ads coming through  https://imgur.com/XV6C24b

**Comment by zGato** [comment]
Well, as I said 😂 

![Image](https://github.com/user-attachments/assets/996b73d3-21c7-433c-ab53-dc94f42c38d9)

Good job Brave!

**Comment by ryanbr** [comment]
This would've eventually happened regardless, but not ideal that twitch changed the adblocking.

**Comment by ryanbr** [comment]
In Brave Nightly 1.82.74 or better, no other extensions needed

And manually add in brave://settings/shields/filters (custom filters)

```
   twitch.tv##+js(vaft-ublock-origin)
   twitch.tv##+js(no-fetch-if, edge.ads.twitch.tv)
```

**Comment by ryanbr** [comment]
Do you see twitch blocking 3rd-party access also @pixeltris ? Looks like sec-ch on twitch embeds on Brave

```
https://escharts.com/live/csgo
https://lolesports.com/live/msi/riotgames
```

**Comment by pixeltris** [comment]
Yes it looks like they broke the embeded player under Brave. They are rejecting the `embed` player type when `Brave` is inside of the `sec-ch-ua` header. This impacts both stable Brave (with no scripts active) and nightly Brave (with and without the above filters).

It sounds like some adjustments to https://github.com/brave/brave-core/pull/29913 would be needed to always force the `sec-ch-ua` change on https://gql.twitch.tv/gql requests (which it currently doesn't seem to be doing when the player is embedded).

**Comment by ghost** [comment]
The Brave implementation to remove Brave from the Sec-CH-UA, it is done done per domain, and whatever network request occurs under a site in that domain (1st and 3rd party network requests, scripts or iframes or css, fonts, xhr or whatever) will have the Sec-CH-UA changed in them. So, it is not like an extension where you can set the header to be changed only in the gql.twitch.tv subdomain in every page that has a twitch embedded video and it will automatically work.

Anyone can check the list https://github.com/brave/adblock-lists/blob/master/brave-lists/brave-checks.txt to know if a site is missing and know if brave will be detected or not by Twitch or others. Then I guess the most important is people reporting it, so it gets added, by using Brave's menu -> help -> report a broken site and choose `there's an anti-adblock measure` and wait for the site to get added eventually. 

But (after reporting a site) people can also manually modify the file to get it working while the domain gets added to the list.
`brave-checks.txt` file is located in the `nlpaeekllejnmhoonlpcefpfnpbajbpe` folder in Brave's user-data, and just adding a domain and restarting the browser will work. Of course, the file, as any other component will get overwritten if it gets updated and then the changes are lost, but adding sites again is not difficult.

**Comment by ryanbr** [comment]
Just testing here, with multi streams. Occasionally one stream it locks up for 4-5mins with a endless loading

<img width="3784" height="2004" alt="Image" src="https://github.com/user-attachments/assets/45f4d613-98b3-4d84-b308-f4e67b5c5f45" />
<img width="3820" height="2016" alt="Image" src="https://github.com/user-attachments/assets/1cfe43ca-a4a7-44dc-9090-3c8a34648db6" />

**Comment by pixeltris** [comment]
I'm assuming the browser in the background doesn't have the script active, and the foreground it has the script active?

What I'm guessing is the server is forcing the purple screen thing due to the embeded player and the frequency is inconsistent as they handle these "ads" slightly differently to regular preroll / midroll ads.

What I could do is force the player type to be `site` which should remove the inconsistencies with the embeded player. The `video-swap-new` script has some code to do this (currently unused as `OPT_ACCESS_TOKEN_PLAYER_TYPE` is set to null rather than `site`):

https://github.com/pixeltris/TwitchAdSolutions/blob/69a2e17a5b57a265f45a265fa51f9aeb2ae3887f/video-swap-new/video-swap-new.user.js#L683-L693

*Unrelated to the above;* sounds like you're going with the `brave-checks.txt` thing only by adding all sites which embed the player? I generally wouldn't recommend it as lots of different websites use the embeded player for various reasons. e.g. streamers with their own websites, video game companies, etc. Lots of "niche" websites. IMO it would be better to find a solution that also works when specific urls are requested as mentioned above (https://gql.twitch.tv/gql), regardless of the current domain.

**Comment by ryanbr** [comment]
Correct the top window in each is where the window has no twitch scriptlet loaded. The bottom has video-swap-new-ublock-origin.js

The current limitation of brave-checks/sec-ch-ua fixes, it applys to the top level domain only, and not embed options. Have enough of the popular twitch 3p sites in use. Why I added so many 3p twitch sites.

**Comment by ghost** [comment]
It will be improved, so it will work as expected someday (hopefully soon) https://github.com/brave/brave-browser/issues/47688
The current implementation is better than nothing or better than using extensions, especially since Android can't use extensions.
Brave-checks should be enough and cover most cases for most people, until iframes get the sec-ch-ua properly and not just the top level domain.

**Comment by ryanbr** [comment]
`Your browser encountered an error while decoding the video. (Error #3000)`

@pixeltris  has twitch updated the ad injection?

**Comment by ryanbr** [comment]
Any updates @pixeltris ?

**Comment by pixeltris** [comment]
I think error 3000 was from 2k/4k streams - see https://github.com/pixeltris/TwitchAdSolutions/issues/447#issuecomment-3476564757. The latest version of the handles 2k/4k streams so hopefully error 3000/4000 shouldn't happen. See https://github.com/pixeltris/TwitchAdSolutions/issues/451#issuecomment-3476568198

I've changed the script to set the player type to `site` when embeded on other websites which addresses https://github.com/pixeltris/TwitchAdSolutions/issues/282#issuecomment-3078678195

A buffering mitigation was added in the latest version of the script (it does a pause/play during ads when buffering occurs, this is to prevent infinite buffering that sometimes happens).

I think that's all the updates.

It sounds like Twitch wasn't too happy when the script was enabled by default under Brave a little while ago. You guys might want to consider if the script is suitable. It's pretty spam heavy in terms of requests. This is something they complained about in 2019. See https://github.com/uBlockOrigin/uAssets/issues/5184#issuecomment-475864036

**Comment by pixeltris** [comment]
Think this issue has run its course. Best of luck.
